ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -1031,7 +1040,11 @@ func (r *RayJobReconciler) checkSubmitterAndUpdateStatusIfNeeded(ctx context.Con
 		}
 
 		if headPod == nil {
-			logger.Info("Ray head pod not found, skipping sidecar container status check")
+			// If head pod is deleted, mark the RayJob as failed
+			shouldUpdate = true
```

Comment:
There is no point of setting `shouldUpdate = true` here since there's an early return on line 1048

Reply:
Oh, it is needed since `shouldUpdate` is one of the return values.